File: thrifty-schema/src/test/java/com/microsoft/thrifty/schema/BuiltinTypeTest.java
package com.microsoft.thrifty.schema;

import org.junit.Test;

import static org.hamcrest.Matchers.is;
import static org.junit.Assert.assertThat;

public class BuiltinTypeTest {
    @Test
    public void boolIsNotNumeric() {
        assertThat(((BuiltinType) BuiltinType.BOOL).isNumeric(), is(false));
    }

    @Test
    public void byteIsNumeric() {
        assertThat(((BuiltinType) BuiltinType.BYTE).isNumeric(), is(true));
    }

    @Test
    public void i8IsNumeric() {
        assertThat(((BuiltinType) BuiltinType.I8).isNumeric(), is(true));
    }

    @Test
    public void i16IsNumeric() {
        assertThat(((BuiltinType) BuiltinType.I16).isNumeric(), is(true));
    }

    @Test
    public void i32IsNumeric() {
        assertThat(((BuiltinType) BuiltinType.I32).isNumeric(), is(true));
    }

    @Test
    public void i64IsNumeric() {
        assertThat(((BuiltinType) BuiltinType.I64).isNumeric(), is(true));
    }

    @Test
    public void doubleIsNumeric() {
        assertThat(((BuiltinType) BuiltinType.DOUBLE).isNumeric(), is(true));
    }

    @Test
    public void stringIsNotNumeric() {
        assertThat(((BuiltinType) BuiltinType.STRING).isNumeric(), is(false));
    }

    @Test
    public void binaryIsNotNumeric() {
        assertThat(((BuiltinType) BuiltinType.BINARY).isNumeric(), is(false));
    }

    @Test
    public void voidIsNotNumeric() {
        assertThat(((BuiltinType) BuiltinType.VOID).isNumeric(), is(false));
    }
}

File: thrifty-schema/src/test/kotlin/com/microsoft/thrifty/schema/BuiltinTypeTest.kt
/*
 * Thrifty
 *
 * Copyright (c) Microsoft Corporation
 *
 * All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the License);
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING
 * WITHOUT LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE,
 * FITNESS FOR A PARTICULAR PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
 *
 * See the Apache Version 2.0 License for specific language governing permissions and limitations under the License.
 */
package com.microsoft.thrifty.schema

import io.kotest.matchers.shouldBe
import org.junit.Test

class BuiltinTypeTest {
    @Test
    fun boolIsNotNumeric() {
        (BuiltinType.BOOL as BuiltinType).isNumeric shouldBe false
    }

    @Test
    fun byteIsNumeric() {
        (BuiltinType.BYTE as BuiltinType).isNumeric shouldBe true
    }

    @Test
    fun i8IsNumeric() {
        (BuiltinType.I8 as BuiltinType).isNumeric shouldBe true
    }

    @Test
    fun i16IsNumeric() {
        (BuiltinType.I16 as BuiltinType).isNumeric shouldBe true
    }

    @Test
    fun i32IsNumeric() {
        (BuiltinType.I32 as BuiltinType).isNumeric shouldBe true
    }

    @Test
    fun i64IsNumeric() {
        (BuiltinType.I64 as BuiltinType).isNumeric shouldBe true
    }

    @Test
    fun doubleIsNumeric() {
        (BuiltinType.DOUBLE as BuiltinType).isNumeric shouldBe true
    }

    @Test
    fun stringIsNotNumeric() {
        (BuiltinType.STRING as BuiltinType).isNumeric shouldBe false
    }

    @Test
    fun binaryIsNotNumeric() {
        (BuiltinType.BINARY as BuiltinType).isNumeric shouldBe false
    }

    @Test
    fun voidIsNotNumeric() {
        (BuiltinType.VOID as BuiltinType).isNumeric shouldBe false
    }
}